### DISCUSSION TITLE
Delegate ts class field transformation from `strip` to `class_properties`

### DISCUSSION THREAD

**Discussion by Austaras** [swc-project/swc#3562] [General]
Concerning recent issues of #3317 and #3389 which requires [[Define]] for class fields in typescript, here is my proposed steps to fix:
1. fixes bugs in `classes` #3505 so we can reuse babel tests
2. add loose mode to `classes` #3474 
3. fixes bugs in `class_properties` so we can reuse babel tests #3594 #3645 #3665 #3695 
4. add loose mode to `class_properties` #3722 
5. expose `use_define_for_class_field` in `config.jsc`. If this is set to true, automatically set `set_public_class_fields` to true, #3766

**Comment by kdy1** [comment]
cc @dsherret @bartlomieju

**Comment by Austaras** [comment]
This would be a break change for browsers natively support class field, but since tsc doesn't have `env` and babel does the same, I guess it's fine
